### PR TITLE
fix(training): source expert_only supported-policy set live from lerobot

### PR DESCRIPTION
**Pull request by cagataycali**
## Problem

`LerobotTrainer.validate` gated `method="expert_only"` (freeze the (V)LM backbone, train only the action expert) against a **hard-coded** tuple `_EXPERT_ONLY_POLICY_TYPES = {pi0, pi05, pi0_fast, smolvla}`, and the `lerobot_train` tool's `build_train_command` used a **separate** hard-copied set for the same check. This is the stale-allowlist drift class already fixed for `lerobot_async` (#1458) and the reward-model registry.

The copy is not merely a latent landmine — it is **already wrong** against current lerobot:

* lerobot's `PI0FASTConfig` exposes **no** `train_expert_only` field. So an `expert_only` **pi0_fast** run:
  1. passes `validate` (pi0_fast was in the hardcoded allow set), then
  2. `build_config` applies the flag under a `hasattr(cfg, "train_expert_only")` guard → the field is **absent** → the flag is **silently skipped** → the run **full-finetunes the entire backbone** (a far more expensive, different run) while reporting success.
* Conversely, a policy lerobot later *gains* the field would be wrongly **rejected** by the frozen copy.

Empirically on the installed lerobot the configs that actually declare `train_expert_only` are `{pi0, pi05, smolvla}` — **not** the hardcoded four.

## Fix

Source the supported set **live**: a policy supports `expert_only` **iff** its lerobot config class declares a `train_expert_only` field (`dataclasses.fields`).

* `strands_robots/training/lerobot.py`: new `_policy_supports_expert_only(ptype)` (via the existing live `_policy_registry()`); `validate` now rejects `expert_only` for a policy whose config lacks the field, naming the live supported set; the reused `_EXPERT_ONLY_POLICY_TYPES` becomes an offline **fallback** only.
* `strands_robots/tools/lerobot_train.py`: `_expert_only_policy_types()` sources live from `PreTrainedConfig.get_known_choices()`, with a static `_EXPERT_ONLY_POLICIES_FALLBACK` used only when lerobot/`get_known_choices` is unimportable; `build_train_command` uses it.
* Docstrings no longer claim a fixed `pi0/pi05/pi0_fast/smolvla` list.

Backward compatible: the currently-supported policies (`pi0`, `pi05`, `smolvla`) behave identically; only the incorrectly-listed `pi0_fast` is now (correctly) rejected up front instead of silently dropping the flag.

## Tests (fail before, pass after)

`tests/training/test_lerobot.py::TestExpertOnlyMethod` and `tests/tools/test_lerobot_train.py`:

* `validate` rejects `expert_only` for pi0_fast (**fails-before**: returned `[]`), accepts it for pi0/pi05/smolvla.
* `build_train_command(policy_type="pi0_fast", train_expert_only=True)` raises (**fails-before**: built the argv fine, since pi0_fast was in the hardcoded set).
* **Drift guards** (version-agnostic, skip offline): `_policy_supports_expert_only` tracks each config's actual `train_expert_only` field, and the tool's live set equals lerobot's `train_expert_only`-declaring configs.

Gate: `ruff check`/`format` clean, `mypy` clean on both changed source files, 218 passed across `test_lerobot.py` + `test_lerobot_train.py` + `test_lerobot_policy_type_discovery.py` + `test_lerobot_train_blocklist.py`.

No visual artifact (a training-config validation/argv fix, no policy/sim/render/asset change; the fails-before tests are the proof).

---

### Round 2 changelog

* **Fix code-scanning finding** (`Potentially uninitialized local variable PreTrainedConfig`, `tests/tools/test_lerobot_train.py`): the drift-guard test bound `PreTrainedConfig` inside a `try` and used it after the `try/except`, where the `except` calls `pytest.skip()`. The analyzer does not model `pytest.skip()` as terminating, so it saw a path reaching the comprehension with `PreTrainedConfig` unbound. Switched to `pytest.importorskip` (the dominant idiom in this suite), which still performs the `lerobot.policies` side-effect registration import and skips offline, while binding `PreTrainedConfig` unconditionally. Offline behavior (skip) is unchanged; no source/behavior change. (`ff54ffb`)

### CODE REVIEW

**Review by yinsong1986 — DISMISSED** [review]
## Summary

Replaces two independently hard-coded `expert_only` policy allowlists (`LerobotTrainer.validate` and the `lerobot_train` tool's `build_train_command`) with live capability probes: a policy supports `method="expert_only"` iff its lerobot config class declares a `train_expert_only` dataclass field. This is the same stale-allowlist-drift fix pattern already applied for `lerobot_async` (#1458), and it corrects a real existing bug — `pi0_fast` was in both hardcoded sets but its lerobot config has no `train_expert_only` field, so an `expert_only` pi0_fast run passed validation and then silently full-finetuned the whole backbone via the `hasattr` guard in `build_config`. Verified live against the installed lerobot: the `train_expert_only`-declaring configs are exactly `{pi0, pi05, smolvla}`, matching the new fallback snapshots. The new probe mirrors the existing `_policy_supports_relative_actions` shape (class-level field lookup via the shared `_policy_registry()`, no instantiation, documented offline fallback), fails loudly with an actionable `ValueError`/validate problem naming the live supported set, and the fails-before tests plus version-agnostic drift guards pin the behavior.

## What's good

- Fails-before tests on both surfaces (`validate` rejection for pi0_fast, `build_train_command` raise) plus drift-guard tests that compare the live set against lerobot's actual registry, keeping the fallback honest.
- The behavior change for pi0_fast is loud (explicit rejection with the supported set named) rather than a silent default — consistent with AGENTS.md's no-silent-defaults rule — and is documented in the CHANGELOG.
- Consistent with the established live-registry pattern in `strands_robots/training/lerobot.py` (`_policy_registry`, `_policy_supports_relative_actions`), so there is now one discovery mechanism rather than a third copy.

## Verification suggestions

- `python -c "import dataclasses; import lerobot.policies; from lerobot.configs.policies import PreTrainedConfig; print(sorted(n for n,c in PreTrainedConfig.get_known_choices().items() if any(f.name=='train_expert_only' for f in dataclasses.fields(c))))"` — confirms the live set is `['pi0', 'pi05', 'smolvla']` on the pinned lerobot (reproduced locally during this review).
- `pytest tests/training/test_lerobot.py -k expert -q` and `pytest tests/tools/test_lerobot_train.py -q` — 8 and 42 passed respectively against the PR head in a clean worktree.

File: tests/tools/test_lerobot_train.py [hunk 1]
```diff
@@ -162,6 +162,38 @@
...
+    import dataclasses
+
+    try:
+        import lerobot.policies  # noqa: F401
+        from lerobot.configs.policies import PreTrainedConfig
+    except Exception:  # noqa: BLE001
+        pytest.skip("lerobot registry unavailable offline")
+    expected = {
+        name
+        for name, cfg_cls in PreTrainedConfig.get_known_choices().items()
```
[the remark this answers is not in the document]
Comment (cagataycali) [reply tests/tools/test_lerobot_train.py:189]: Fixed in `ff54ffb`. The prior `try/except` bound `PreTrainedConfig` inside the `try` and used it after, with the `except` calling `pytest.skip()`; the analyzer does not model `pytest.skip()` as terminating, so it saw a path reaching the comprehension with `PreTrainedConfig` unbound. Switched to `pytest.importorskip` (the dominant idiom in this suite), which still performs the `lerobot.policies` side-effect registration import and skips offline, while binding `PreTrainedConfig` unconditionally on any path that reaches the use. Offline behavior (skip) is unchanged; no source/behavior change.

**Review by yinsong1986 — APPROVED** [review]
## Summary

Replaces the two independently hard-coded `expert_only` allowlists (`LerobotTrainer.validate` and `build_train_command`) with live capability probes: a policy supports `method="expert_only"` iff its lerobot config class declares a `train_expert_only` dataclass field, with the static snapshot demoted to an offline-only fallback. This mirrors the pattern already in place for `_policy_supports_relative_actions` and the #1458 `lerobot_async` fix, and it corrects a real live bug: an `expert_only` pi0_fast run previously passed validation and then silently full-finetuned the backbone because `build_config`'s `hasattr` guard skipped the absent field. I independently probed the installed lerobot registry and confirmed the `train_expert_only`-declaring set is exactly `{pi0, pi05, smolvla}` with `pi0_fast` registered but lacking the field, matching the PR's claim; the new fallback snapshots are faithful to that. The behavior change (rejecting pi0_fast up front) surfaces as an explicit `ValueError`/validate problem rather than a silent drop, so it is the fix, not a silent break. The prior CodeQL uninitialized-variable finding on the drift-guard test is resolved at this head via `pytest.importorskip`, and `Analyze (Python)` is green.

## What's good

- Fails-before regression tests plus version-agnostic drift guards that assert the live set against lerobot's registry directly, so the fallback can't silently rot.
- Verified locally at head `ff54ffbd`: all `TestExpertOnlyMethod` + `test_lerobot_train.py` expert-only tests pass with a real lerobot install; `ruff check`/`ruff format --check` clean on all four changed files; no non-ASCII, no host paths.
- CHANGELOG entry and docstring updates land in the same PR; scope is tightly limited to the stated fix.
